File: repositories/admin_action_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type AdminActionRepository interface {
	Create(ctx context.Context, action *entities.AdminAction) error
	FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.AdminAction, error)
}

type GormAdminActionRepository struct {
	db *gorm.DB
}

func NewAdminActionRepository(db *gorm.DB) AdminActionRepository {
	return &GormAdminActionRepository{db: db}
}

func (r *GormAdminActionRepository) Create(ctx context.Context, action *entities.AdminAction) error {
	return r.db.WithContext(ctx).Create(action).Error
}

func (r *GormAdminActionRepository) FindByDoctorID(ctx context.Context, doctorID uuid.UUID) ([]entities.AdminAction, error) {
	var actions []entities.AdminAction
	err := r.db.WithContext(ctx).Where("doctor_id = ?", doctorID).Order("created_at DESC").Find(&actions).Error
	return actions, err
}

File: python_backend/repositories/admin_action_repository.py
from typing import List
from uuid import UUID
from sqlalchemy.orm import Session
from entities.admin_action import AdminAction

class AdminActionRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, action: AdminAction) -> None:
        self.db.add(action)
        self.db.commit()
        self.db.refresh(action)

    def FindByDoctorID(self, doctor_id: UUID) -> List[AdminAction]:
        return self.db.query(AdminAction).filter(AdminAction.doctor_id == doctor_id).order_by(AdminAction.created_at.desc()).all()
